internal/template-bundle/bundle.go

```diff
@@ -0,0 +1,101 @@
+package template_bundle
+
+import (
+	"bytes"
+	"io"
+	"io/ioutil"
+
+	templatev1 "github.com/openshift/api/template/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/yaml"
+
+	cdiv1beta1 "kubevirt.io/containerized-data-importer-api/pkg/apis/core/v1beta1"
+)
+
+type Bundle struct {
+	Templates   []templatev1.Template
+	DataSources []cdiv1beta1.DataSource
+}
+
+func ReadBundle(filename string) (Bundle, error) {
+	templates, err := readTemplates(filename)
+	if err != nil {
+		return Bundle{}, err
+	}
+
+	return Bundle{
+		Templates:   templates,
+		DataSources: extractDataSources(templates),
+	}, nil
+}
+
+func readTemplates(filename string) ([]templatev1.Template, error) {
+	var bundle []templatev1.Template
+	file, err := ioutil.ReadFile(filename)
+	if err != nil {
+		return nil, err
+	}
+	decoder := yaml.NewYAMLOrJSONDecoder(bytes.NewReader(file), 1024)
+	for {
+		template := templatev1.Template{}
+		err = decoder.Decode(&template)
+		if err == io.EOF {
+			return bundle, nil
+		}
+		if err != nil {
+			return nil, err
+		}
+		if template.Name != "" {
+			bundle = append(bundle, template)
```

Comment:
Would be possible to combine `readTemplates` and `extractDataSources` into one function (to not iterate through the template slice twice)?

Reply:
Yes, would be possible, but I think it would be less clear code.
The way it is now, the two functions do only one thing. Combining them would be a little faster, but more complex code. And more performance is not important, because the templates are loaded only once when the operator starts.